Activate annotation processors again for panache

Postremus opened the pull request:
We depend on the jpamodelgen annotation processor in the panache modules, to create the Meta Model for e.g. the PanacheEntity class.

closes #14926

Postremus commented:
@stuartwdouglas
Would you please review this PR, please?
According to git blame, you added the proc:none initially.

famod commented:
FTR, `proc:none` was added in this commit https://github.com/quarkusio/quarkus/commit/a28d76b436f0829e332c63e6a5d320d7eebd35f5 (1.5.0.CR1).

stuartwdouglas reviewed (changes requested):
We don't want PanacheAnnotationProcessor (https://github.com/quarkusio/quarkus/commit/a28d76b436f0829e332c63e6a5d320d7eebd35f5#diff-3b62cdfb1058fa132560380d194f6b00371f9426fdc7f4bf759a2ccb36c7febaR28) to be run, can you change this to just run the hibernate one?

Postremus commented:
@stuartwdouglas done

stuartwdouglas commented:
Can we also add a test for this to make sure that we don't run into this in future? Otherwise looks good.

Postremus commented:
@stuartwdouglas Added a simple test to hibernate-panache and hibernate-reactive-panache, which just checks if the metamodel was generated.

Postremus committed:
Activate annotation processors again for panache
We depend on the jpamodelgen annotation processor in the panache modules, to create the Meta Model for e.g. the PanacheEntity class.

famod commented:
Thanks @Postremus!

ninaHerpertz commented:
I am trying to get generated model classes, but I have to extend the PanacheEntityBase class as I need a different generation strategy for my ID field. The generation only works on PanacheEntity classes as far as I have seen. Is there a way to extend the generation to the PanacheEntityBase classes or is there any other way to do this?

Postremus commented:
@ninaHerpertz Simply adding the hibernate-jpamodelgen annotation processor should work in your case. See https://quarkus.io/guides/hibernate-orm-panache#defining-entities-in-external-projects-or-jars

You need to do this anyway, even for PanacheEntity.
Quarkus uses jpamodelgen for meta-model generation inside the quarkus-hibernate-orm-panache module. Any other modules containing entities need to define this annotation processor seperatly.
